File: lab_inject.py
import sys
import sqlite3 as sql

def main():
	sqliteFile = 'pokedex.sqlite'

	try:
		con = sql.connect(sqliteFile)
		cur = con.cursor()

		print('Find all the pokemon related to ______')
		userInput = input('>')

		cur.execute('''SELECT a.identifier
						FROM pokemon_species AS a
						LEFT JOIN pokemon_species AS b
    						ON a.evolution_chain_id = b.evolution_chain_id
						WHERE b.identifier = LOWER("%s");''' % userInput )

		for name, in cur:
			print(name, end=', ')
		print()

	except TypeError:
		print('Couldn\'t find %s in database' % userInput)      

	except sql.Error as e:
		print("Error %s:" % e.args[0])

	finally:
		if con:
			con.close()

if __name__ == '__main__':
	sys.exit(main())

File: lab_inject.cpp
#include <iostream>
#include <string>
using namespace std;

#include "libsqlite.hpp"

int main()
{
    try
    {
        sqlite::sqlite db( "pokedex.sqlite" );
        
        cout << "Find all the pokemon related to ______" << endl;

        string userInput;
        getline( cin, userInput );

        auto cur = db.get_statement();
        cur->set_sql( "SELECT a.identifier "
                      "FROM pokemon_species AS a "
                      "LEFT JOIN pokemon_species AS b "
                      "  ON a.evolution_chain_id = b.evolution_chain_id "
                      "WHERE b.identifier = LOWER(\"" + userInput + "\");" );
        cur->prepare();
    
        while( cur->step() )
            cout << cur->get_text(0) << ", ";
        cout << endl;
    }
    catch( sqlite:: exception e )
    {
        cerr << e.what() << endl;
        return 1;
    }

    return 0;
}